Example-CustomUserGroups-REXX/README.md

```diff
@@ -0,0 +1,498 @@
+## REXX Custom User Group Sample
+
+This REXX sample utilizes the **HWIREST API** to do one of the following:
+- List the Custom User Groups on the specified CPC
+- List the members of the specified Custom User Group on the specified CPC
+- Attempts to add the specified LPAR as a new member to the specified Custom User Group on the specified CPC
+- Attempts to remove the specified LPAR as a member from the specified Custom User Group on the specified CPC
+
+## System Prep work
+- Store USRGRP01 into a data set of at least fb80 and at most fb255
+- Ensure your z/OS user ID has at least READ and CONTROL access to the following FACILITY Class Profile
+    - HWI.TARGET.netid.nau
+
+    <p>where netid.nau represents the 3– to 17– character SNA name of the particular CPC. </p>
+
+## Invocation
+**Syntax**:
+```
+USRGRP01 [-C CPCName] [-L LPARName] [-G CustomUserGroup] [-A Action {ADD, REMOVE, LIST}] [-I] [-V] [-H]
+```
+
+ Optional Input Parameters:<br>
+  - CPCName  - The name of the CPC to query, default is the local CPC<br>
+  - LPARName - The name of the LPAR to query and target, default is
+                 the local LPAR<br>
+  - UserGroupName - The name of the Custom User Group to target<br>
+  - Action        - The action to take against the Custom User Group
+                    limited to ADD, REMOVE, or LIST<br>
+  - -I            - indicate running in ISV REXX environment, default
+                    is TSO/E<br>
+  - -V            - turn on additional verbose JSON tracing<br>
+  - -H            - Display sample parameters and how to invoke the sample<br>
+
+**Note**: If no arguments are supplied then the sample will default to listing all of the Custom User Groups on the local CPC.
+
+**Sample Invocation in TSO:**
+
+USRGRP01 has been copied into data set HWI.USER.REXX <br>
+
+  List all Custom User Groups on the specified CPC:<br>
+  - Local CPC and LPAR:
+    ```
+    ex 'HWI.USER.REXX(USRGRP01)'
+    ```
+
+  - Remote CPC and LPAR:
+    ```
+    ex 'HWI.USER.REXX(USRGRP01)' '-C CPC1 -L LPAR1'
+    ```
+
+  List the members of the TEST Custom User Group:
+  - Local CPC and LPAR:
+    ```
+    ex 'HWI.USER.REXX(USRGRP01)' '-G TEST'
+    ```
+
+  - Remote CPC and LPAR:
+    ```
+    ex 'HWI.USER.REXX(USRGRP01)' '-C CPC1 -L LPAR1 -G TEST'
+    ```
+
+  Add an LPAR as a new member to the TEST Custom User Group
+
+  - Local CPC and LPAR:

```

Comment:
2 things
1. per the HMC spec, a group can contain any variety of objects 
```
A Group object represents one or more managed objects which are called group members. Each member
is of some object type: CPC, Logical Partition, etc.
```
so if a user does a -A ADD, you're only adding the local LPAR to the group? not the local CPC?

2. there is a high probability someone may accidentally leave off the -L or -C when driving this...is it extremely clear  from the output that the user asked for the local lpar or local cpc to be added to the group?

Reply:
1. Correct, this sample is simply designed to add the specified LPAR to the target Custom User Group if the appropriate arguments are specified (I described this functionality in the README as part of the sample's description). 
2. A message is displayed indicating the target CPC's name and the target Custom User Group's name if one was provided 
`Starting LSTGRPS for CPC name:T31 and Customer User Group name:               `

I could modify the message to be more clear if you want